homeassistant/components/evohome/__init__.py

```diff
@@ -98,7 +98,9 @@ def _handle_exception(err):
     if not await broker.init_client():
         return False
 
-    load_platform(hass, 'climate', DOMAIN, {}, hass_config)
+    hass.async_create_task(
+        async_load_platform(hass, 'climate', DOMAIN, {}, hass_config))
+
```

Comment:
hmm, well maybe a mistake to merge the prio PR. Please don't mix async with sync library. If you use sync library, use also our sync core calls. Please rewrite your library to real async or revert the async function stuff

Reply:
We have allready a high optimize wrapper for sync library/interfaces and don't need and want wrapper code outsite this layers otherwise we have 1000x the same wrapper code on lvl1 integrations.

Reply:
@pvizeli & @MartinHjelmare 

I am not sure if I have been given conflicting advice, or if I am simply confused (I do struggle with sync/async), so could I ask for clarification?

The client library is sync only & it is not my library. However, the integration's `setup()` invokes a class (`evoBroker`) that relies upon `self.hass.helpers.storage` (to keep OAuth tokens across HA restarts), which (AFAIKT) is async only.  

Could you be a little more clear about what you need me to change?  I'm happy to change it, I just don't know exactly what you require...

Reply:
Not a problem with config entries, as those are loaded prior to invoking your class 😉

Reply:
this change has been moved to PR #25035